exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentContextImpl.java

```diff
@@ -136,8 +143,8 @@ public void interrupt(final InterruptedException e) {
   private final AccountingUserConnection accountingUserConnection;
   /** Stores constants and their holders by type */
   private final Map<String, Map<MinorType, ValueHolder>> constantValueHolderCache;
-
-  private RuntimeFilterSink runtimeFilterSink;
+  private Map<Long, RuntimeFilterWritable> rfIdentifier2RFW = new ConcurrentHashMap<>();
+  private Map<Long, Boolean> rfIdentifier2fetched = new ConcurrentHashMap<>();
```

Comment:
It can't be replaced by the `rfIdentifier2RFW ` thorough setting the value to null. Since we need to decrease the reference number of the received RFW's Drillbuf to keep reference number count balance, we should keep the RFW object reference to do the decreasing at the `closeReceivedRFWs ` . Further, we need to do the decreasing even it has been consumed to keep reference number balance.

Reply:
make sense, will refactor.

Reply:
Still keep the original implementation. The reason is we fetch the RFW more than one times at the `RuntimeFilterRecordBatch` side. If we remove it from the `rfIdentifier2RFW` map, it will cause NPE.

Reply:
RuntimeFilterRecordBatch should not call get multiple times once it has received a valid RTF. Anyways I guess that's not a big deal and can be refactored in future.